syslog-ng-helm-chart/templates/deployment.yaml

```diff
@@ -0,0 +1,90 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "syslog-ng.fullname" . }}
+  labels:
+    {{- include "syslog-ng.labels" . | nindent 4 }}
+spec:
+{{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+{{- end }}
+  selector:
+    matchLabels:
+      {{- include "syslog-ng.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+    {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      labels:
+        {{- include "syslog-ng.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+          - name: tcp-port
+            containerPort: 601
+            protocol: TCP
+          - name: udp-port
+            containerPort: 514
+            protocol: UDP
+          volumeMounts:
+          - mountPath: /etc/syslog-ng/syslog-ng.conf
+            name: config
+            subPath: syslog-ng.conf
+        {{- if .Values.storage.enable }}
+          - mountPath: /var/log
+            name: logs
+        {{- end }}
+  ###
+        {{- if .Values.logrotate.enable }}
+          - name: logrotate
+            securityContext:
+              {{- toYaml .Values.securityContext | nindent 12 }}
+            image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+            imagePullPolicy: {{ .Values.image.pullPolicy }}
+            command:
+              -"/bin/logrotate"
+              -"--help"
```

Comment:
I not a kubernetes expert, but I don't see why "logrotate --help" makes sense here. When is it executed?

Reply:
Thats a mistake. i was testing a co-container in the same pod, for logrotate but finally decide to use a cronjob, this was never included on the deployment as in values there is no logrotate.enable. I will delete this and commit :D